### PR TITLE
Capitalisation of tribe names - Atlanteans + Frisians + Amazons

### LINKED ISSUE #5041: Capitalisation of Tribe names
Tribe names are not capitalised within the game consistently. I noticed this with the Amazons, but hessenfarmer mentioned it should be the case for all tribes (see [comment 37928 in the forum](https://www.widelands.org/forum/topic/2897/?page=40#post-37928)).

I'll start with Amazons and then go through the other tribes.

### PR DESCRIPTION
**Pull request by jgsogo**
**Type of change**
Refactoring

**Issue(s) closed**
closes #5041: here it fixes Atlanteans, Frisians and Amazons. Previous PRs were fixing [Empire](https://github.com/widelands/widelands/pull/5913) and [Barbarians](https://github.com/widelands/widelands/pull/5910).

**New behavior**

 * I'm modifying:
    * string constants, 
    * comments,
 * I'm NOT changing:
    * `.po` files. I assume they will be regenerated from the sources
    * `Changelog`

### CODE REVIEW

File: src/ai/defaultai.h [hunk 1]
```diff
@@ -383,9 +383,9 @@ struct DefaultAI : ComputerPlayer {
 
 	int16_t productionsites_ratio_;
 
-	bool resource_necessity_water_needed_{false};  // unless atlanteans or amazons
+	bool resource_necessity_water_needed_{false};  // unless the Atlanteans or the Amazons
```
Comment (tothxa) [inline src/ai/defaultai.h]: I don't think this needs the "the"-s. OTOH it should probably be rephrased. @hessenfarmer?

File: data/tribes/initialization/empire/units.lua [hunk 2]
```diff
@@ -1047,92 +1047,92 @@ wl.Descriptions():new_tribe {
       {
          name = "deadtree7",
          helptexts = {
-            -- TRANSLATORS: Helptext for an amazon immovable usable by empire: Dead Tree
+            -- TRANSLATORS: Helptext for an Amazon immovable usable by empire: Dead Tree
```
Comment (tothxa) [inline data/tribes/initialization/empire/units.lua]: Haven't you corrected Empire in these in the other PR? This is what @hessenfarmer warned you about. ;)

Comment (hessenfarmer) [reply data/tribes/initialization/empire/units.lua]: I looked in the other PR and there were some instances of empire left out including this one (on a first look all comments regarding workers, buildings and immovables as well.)
so please fix them in this PR as well, as this shall be the last one to fix the issue.

File: src/ai/defaultai.cc [hunk 3]
```diff
@@ -1729,7 +1729,7 @@ void DefaultAI::update_buildable_field(BuildableField& field) {
 			   find_open_water);
 		}
 
-		if (resource_necessity_water_needed_) {  // for atlanteans
+		if (resource_necessity_water_needed_) {  // for the Atlanteans
```
Comment (hessenfarmer) [inline src/ai/defaultai.cc]:
```suggestion
		if (resource_necessity_water_needed_) {  // for the Atlanteans and the Amazons
```

File: src/ai/defaultai.h (hunk 1 above)
Comment (hessenfarmer) [reply src/ai/defaultai.h]:
```suggestion
	bool resource_necessity_water_needed_{false};  // unless we are Atlanteans or Amazons
```

**Review by hessenfarmer — COMMENTED** [review]
some inliners,
after fixing them and the missing empire occurences this might be good to go

File: data/tribes/initialization/empire/units.lua (hunk 2 above)
Comment (jgsogo) [reply data/tribes/initialization/empire/units.lua]: I think I just overlooked this file in #5913. Fixed!

**Review by hessenfarmer — APPROVED** [review]
LGTM now thanks a lot

**Pull request merged**